src/server/routes/ssoProviders/googleSso.js

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021-2022 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Defines constants and functions to handle the SSO workflow using Google as
+ * the provider.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/fluid-project/preferencesServer/blob/main/LICENSE
+ */
+
+"use strict";
+
+const axios = require("axios");
+const generateNonce = require("../../generateRandomToken.js");
+
+const REDIRECT_URI = "http://localhost:3000/sso/google/login/callback";
+
+const options = {
+    // Google authorization endpoint to start their authorization workflow.
+    authorizeUri: "https://accounts.google.com/o/oauth2/auth",
+
+    // Google "get access token" and "refresh access token" endpoints.
+    accessTokenUri: "https://accounts.google.com/o/oauth2/token",
+    accessType: "offline",
+
+    // Google user endpoints.
+    userInfoUri: "https://www.googleapis.com/oauth2/v2/userinfo",
+
+    // Google "use access token endpoint"
+    credentialsUri: "https://oauth2.googleapis.com/token",
+
+    // URI that Google uses to call back to the Personal Data Server for
+    // logging in.
+    redirectUri: REDIRECT_URI,
+    encodedRedirectUri: encodeURIComponent(REDIRECT_URI),
+
+    // Identifier for retrieving the client information (e.g. id and secret)
+    // for the Personal Data Server database.
+    provider: "google"
+};
+
+class GoogleSso {
```

Comment:
As @klown mentioned at today's meeting, APIs provided by different OAuth 2 providers may differ at http requests and responses. It would be better to wait until implementing next OAuth 2 workflow from another provider to better understand how to generalize SSO implementation.